clients/venice-admin-tool/src/main/java/com/linkedin/venice/TopicMessageFinder.java

```diff
@@ -143,4 +150,92 @@ public static byte[] serializeKey(String keyString, String keySchemaStr) {
 
     return keySerializer.serialize(key);
   }
+
+  protected static KeyPartitionInfo findPartitionIdForKey(
+      ControllerClient controllerClient,
+      String storeName,
+      int versionNumber,
+      String key,
+      String keySchemaStr) {
+    StoreResponse storeResponse = controllerClient.getStore(storeName);
+    if (storeResponse == null) {
+      throw new VeniceNoStoreException("Store " + storeName + " does not exist.");
+    }
+    StoreInfo storeInfo = storeResponse.getStore();
+    if (storeInfo == null) {
+      throw new VeniceNoStoreException("Store " + storeName + " does not exist.");
+    }
+
+    Optional<Version> versionInfo = storeInfo.getVersion(versionNumber);
+
+    int partitionCount;
+    PartitionerConfig partitionerConfig;
+    if (versionInfo.isPresent()) {
+      Version version = versionInfo.get();
+      LOGGER.info(
+          "Found store: {} version: {}. Will use partitioner config: {} and partition count: {} from this version",
+          storeName,
+          versionNumber,
+          version.getPartitionerConfig(),
+          version.getPartitionCount());
+      partitionerConfig = version.getPartitionerConfig();
+      partitionCount = version.getPartitionCount();
+    } else {
+      LOGGER.info(
```

Comment:
Should we just throw instead? As this is unexpected state?

Reply:
Version is actually optional arg for the new command so this is not really expected. Also, throwing an exception will affect the current behavior if input topic is rt for topic message finder

Reply:
I see, makes sense to me